velox/functions/lib/window/tests/WindowTestBase.cpp

```diff
@@ -108,14 +182,45 @@ void WindowTestBase::testWindowFunction(
     const std::string& function,
     const std::vector<std::string>& overClauses,
     const std::vector<std::string>& frameClauses,
-    bool createTable) {
+    bool createTable,
+    WindowStyle windowStyle) {
   if (createTable) {
     createDuckDbTable(input);
   }
+
   for (const auto& overClause : overClauses) {
-    for (auto& frameClause : frameClauses) {
-      auto queryInfo =
-          buildWindowQuery(input, function, overClause, frameClause);
+    WindowTestBase::QueryInfo queryInfo;
+    for (const auto& frameClause : frameClauses) {
+      switch (windowStyle) {
+        case WindowStyle::SORT:
+          queryInfo =
+              buildWindowQuery(input, function, overClause, frameClause);
+          break;
+        case WindowStyle::STREAMING:
+          queryInfo = buildStreamingWindowQuery(
+              input, function, overClause, frameClause);
+          break;
+        case WindowStyle::RANDOM: {
+          // Generate a random boolean to determine the window style
+          std::random_device rd;
+          std::mt19937 gen(rd());
+          std::uniform_int_distribution<int> dis(0, 1);
+
+          bool randomWindowStyle = dis(gen);
```

Comment:
It would be simpler to remove the switch statement for windowStyle and write as follows

auto resolvedWindowStyle = windowStyle == RANDOM ? dis(rd()) : windowStyle;
if (resolvedWindowStyle == SORT) {
...
} else {
// Code for streaming window.
}

If we want to add a new windowBuild it would be easier to modify the above conditional.

Reply:
Updated.